BUILD.bazel

```diff
@@ -1,47 +1,16 @@
 package(default_visibility = ["//visibility:public"])
 
-# TODO(jelbourn): figure out if these workarounds are still needed
+load("@build_bazel_rules_nodejs//:defs.bzl", "node_modules_filegroup")
 
-# This rule belongs in node_modules/BUILD
-# It's here as a workaround for
-# https://github.com/bazelbuild/bazel/issues/374#issuecomment-296217940
+# Since the node binary rules will preserve the symlinks, we need to reference all node modules
+# that are being used (also dependencies of a used module). This might slow-down running node
+# binaries (e.g. packaging, scss bundling), but it also ensures hermeticity.
+# https://bazelbuild.github.io/rules_nodejs/node/node_repositories.html#overview
 filegroup(
   name = "node_modules",
-  # Performance workaround: list individual files
-  # Reduces the number of files as inputs to nodejs_binary:
-  # bazel query "deps(:node_modules)" | wc -l
-  # This won't scale in the general case.
-  # TODO(alexeagle): figure out what to do
-  srcs = glob(["/".join(["node_modules", pkg, "**", ext]) for pkg in [
-    "@angular",
-    "@angular-devkit",
-    "@schematics",
-    "@types",
-    "bytebuffer",
-    "hammerjs",
-    "jasmine",
-    "minimist",
-    "moment",
-    "parse5",
-    "protobufjs",
-    "protractor",
-    "reflect-metadata",
-    "rxjs",
-    "tsickle",
-    "tslib",
-    "tslint",
-    "typescript",
-    "zone.js",
-  ] for ext in [
-    "*.js",
-    "*.json",
-    "*.d.ts",
-  ]] + [
-    "node_modules/http-server/**",
-  ]),
+  srcs = glob(["node_modules/**/*"])
```

Comment:
Due to the new option `preserve-symlinks` (which is used in `angular/angular` as well), dependencies that are not referenced in that `filegroup` can't be used in `nodejs_binary`'s anymore. We could technically also list every dependency like [here](https://github.com/angular/angular/blob/master/BUILD.bazel#L18), but as per [this comment](https://github.com/angular/angular/pull/24881#discussion_r202704619) it seems to be okay to do it that way